### PR TITLE
feat(bidi): support set neutral string

### PR DESCRIPTION
**Pull request by qinshijing**
### Description of the feature or fix

When switching languages ​​externally, take the initiative to reset the neutral string. https://github.com/lvgl/lvgl/issues/6145

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/misc/lv_bidi.c [hunk 1]
```diff
@@ -364,7 +370,11 @@ static bool lv_bidi_letter_is_rtl(uint32_t letter)
 static bool lv_bidi_letter_is_neutral(uint32_t letter)
 {
     uint16_t i;
-    static const char neutrals[] = " \t\n\r.,:;'\"`!?%/\\-=()[]{}<>@#&$|";
+    const char * neutrals = " \t\n\r.,:;'\"`!?%/\\-=()[]{}<>@#&$|";
+    if(custom_neutrals) {
+        neutrals = custom_neutrals;
+    }
+
```
Comment (kisvegabor) [inline src/misc/lv_bidi.c:377]: This way we cannot easily revert to the default.  I suggest
```suggestion
    const char * default_neutrals = " \t\n\r.,:;'\"`!?%/\\-=()[]{}<>@#&$|";
    if(custom_neutrals) {
        neutrals = custom_neutrals;
    } else  {
       neutrals =  default_neutrals;
    }

```

**Review by kisvegabor — COMMENTED** [review]
Thank you for the PR, but first I would like to understand what the root of the issue is. 
See [here](https://github.com/lvgl/lvgl/issues/6145#issuecomment-2084798028).

File: src/misc/lv_bidi.c (hunk 1 above)
Comment (qinshijing) [reply src/misc/lv_bidi.c:377]: The current method is that as long as custom_neutrals is set to NULL, the default neutrals will be used.

Comment (kisvegabor) [reply src/misc/lv_bidi.c:377]: In Line 375 the original `neutrals` are overwritten with `custom_neutrals`, so there is no way to get back the default settings anymore.

Comment (qinshijing) [reply src/misc/lv_bidi.c:377]: > neutrals

I don't quite understand, neutrals is no longer a static pointer, why there is no way to get back the default settings anymore?

Comment (kisvegabor) [reply src/misc/lv_bidi.c:377]: I was stupid, sorry :smile: Somehow I thought it's  `static const char * neutrals = ...`.

**Comment by kisvegabor** [comment]
I made some more research an it turned out that it's really language specific. So let's go with your suggestion.

**Review by kisvegabor — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**